Tests/RevenueCatUITests/PackageFilteringTests.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  PackageFilteringTests.swift
+//  
+//
+//  Created by Nacho Soto on 7/13/23.
+//
+
+import Nimble
+import RevenueCat
+@testable import RevenueCatUI
+import XCTest
+
+@available(iOS 16.0, macOS 13.0, tvOS 16.0, watchOS 9.0, *)
+class PackageFilteringTests: TestCase {
+
+    func testFilterNoPackages() {
+        expect(PaywallData.filter(packages: [], with: [.monthly])) == []
+    }
+
+    func testFilterPackagesWithEmptyList() {
+        expect(PaywallData.filter(packages: [Self.monthly], with: [])) == []
+    }
+
+    func testFilterOutSinglePackge() {
+        expect(PaywallData.filter(packages: [Self.monthly], with: [.annual])) == []
+    }
+
+    func testFilterOutNonSubscriptions() {
+        expect(PaywallData.filter(packages: [Self.consumable], with: [.custom])) == []
+    }
+
+    func testFilterByPackageType() {
+        expect(PaywallData.filter(packages: [Self.monthly, Self.annual], with: [.monthly])) == [Self.monthly]
```

Comment:
with these being lists, perhaps we should also test behavior for repeated values. 
Also I'd add one case where we have more than one value being returned, so that if we had code that looked for `first`, a test would fail